client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/pattern-page-preview-list.tsx

```diff
@@ -40,28 +48,49 @@ const PatternPagePreviewList = ( {
 		[ selectedHeader, selectedSections, selectedFooter ]
 	);
 
+	const handleClick = ( patterns: Pattern[], pageSlug: string ) => {
+		setZoomedPage( patterns );
+		recordTracksEvent( PATTERN_ASSEMBLER_EVENTS.SCREEN_PAGES_PAGE_PREVIEW_CLICK, {
+			pattern_names: patterns.map( ( pattern ) => pattern.name ).join( ',' ),
+			page_slug: pageSlug,
+		} );
+	};
+
 	return (
-		<div className="pattern-assembler__preview-list">
-			<PatternPagePreview
-				title={ translate( 'Homepage' ) }
-				style={ patternPagePreviewStyle }
-				patterns={ homepage }
-				shouldShufflePosts={ isNewSite }
-			/>
-			{ pages.map( ( page ) => (
+		<>
+			<Composite { ...composite } role="listbox" className="pattern-assembler__preview-list">
 				<PatternPagePreview
-					key={ page.ID }
+					composite={ composite }
+					title={ translate( 'Homepage' ) }
 					style={ patternPagePreviewStyle }
-					title={ page.title }
-					patterns={ [
-						...( selectedHeader ? [ selectedHeader ] : [] ),
-						page,
-						...( selectedFooter ? [ selectedFooter ] : [] ),
-					] }
-					shouldShufflePosts={ false }
+					patterns={ homepage }
+					shouldShufflePosts={ isNewSite }
+					onClick={ ( patterns ) => handleClick( patterns, 'homepage' ) }
 				/>
-			) ) }
-		</div>
+				{ pages.map( ( page, index ) => (
+					<PatternPagePreview
+						key={ page.ID }
+						composite={ composite }
+						style={ patternPagePreviewStyle }
+						title={ page.title }
+						patterns={ [
+							...( selectedHeader ? [ selectedHeader ] : [] ),
+							page,
+							...( selectedFooter ? [ selectedFooter ] : [] ),
+						] }
+						shouldShufflePosts={ isNewSite }
+						onClick={ ( patterns ) => handleClick( patterns, selectedPages[ index ] ) }
+					/>
+				) ) }
+			</Composite>
+			<PatternPagePreviewModal
+				style={ patternPagePreviewStyle }
+				patterns={ zoomedPage }
+				shouldShufflePosts={ isNewSite }
+				isOpen={ !! zoomedPage.length }
```

Comment:
When dismissing the modal, currently we'll see that the modal becomes blank first, then some hundred milliseconds later, the modal will be closed.

Can we make another state that explicitly tracks whether the modal is open or not? Something like

```
	const handleClick = ( patterns: Pattern[], pageSlug: string ) => {
		setZoomedPage( patterns );
		setModalOpen( true );
		...
```
```
const [ isModalOpen, setModalOpen ] = useState( false );
```

```
				...
				isOpen={ isModalOpen }
				onClose={ () => setModalOpen( false ) }
```

I tested the above and I think it's better 😄

Reply:
Seems related to https://github.com/reactjs/react-modal/issues/807

Reply:
I see, that's exactly what is happening here! Not really a blocker, though

Reply:
I merged your suggestion since I don't like that flash either 😁